andela/northstar-backend
# #167727440 Setup Sequelize

### What does this PR do?
- Sets up the application to use Sequelize

### Description of tasks to be completed
- install `sequelize-cli`
- add `sequelize` and `pg-hstore` the application's dependencies
- Add  a `.sequelizerc` file, instructing how Sequelize should be bootstrapped
- Set database credentials in the `.env` file 
- Add migration and "undo migration" scripts to `package.json`

### How should this be manually tested?
-  Clone or download this repository
-  On your terminal, `cd` into the application's directory and run: `git checkout ch-setup-sequelize-167727440`
-  Set the following environment variables in your `.env` file:
```
DEV_DB_USER // database username
DEV_DB_PASSWORD // database password
DEV_DATABASE  //name of your local database
DATABASE_URL  // production database url
```

- If you have not, install [Postgresql](https://www.postgresql.org/download)
- Run `npm run db:migrate`
- Check your database, you should now have both Users and Departments table.
- To undo migration, run `npm run undo:migrate`

### What are the relevant Pivotal Tacker stories?
[#167727440](https://www.pivotaltracker.com/story/show/167727440)
